pwa/src/components/DiagnosisForm.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { ReactElement } from 'react'
+import { List, Checkbox, InputItem, WhiteSpace } from 'antd-mobile'
+import '../scss/DiagnosisForm.scss'
+import * as data from './DiagnosisFields.json'
+
+const CheckboxItem = Checkbox.CheckboxItem
+
+/*
+
+Diagnosis Fields are stored in DiagnosisFields.json in the format:
+
+{
+    "diagnoses" : [
+        {
+            "name": "Name",
+            "id": "id",
+            "text-input-title": "Title",
+            "text-input-type": "type" (i.e. number)
+            "group-title": "Group Title",
+            "group": [
+                { "name": "Name1", "id": "id1" },
+                { "name": "Name2", "id": "id2" }
+            ]
+        },
+        {...}
+    ]
+}
+*/
+
+const diagnosisFields = data.diagnoses
+
+export default function DiagnosisForm(): ReactElement {
+    return (
+        <div id="diagnosis-form-wrapper" data-testid="diagnosis-form-wrapper">
+            {diagnosisFields.map((field) => {
+                const formElements: ReactElement[] = []
+
+                if (field.name != null) {
+                    formElements.push(
+                        <CheckboxItem
+                            id={field.id}
+                            data-testid={field.id}
+                            key={field.id}
+                        >
+                            {field.name}
+                        </CheckboxItem>,
+                    )
+                }
+
+                if (
+                    field['text-input-title'] != null &&
+                    field['text-input-id'] != null
+                ) {
+                    formElements.push(
+                        <List renderHeader={() => field['text-input-title']}>
+                            <InputItem
+                                id={field['text-input-id']}
+                                data-testid={field['text-input-id']}
+                                key={field['text-input-id']}
+                                type={
+                                    field['text-input-type'] === 'number'
+                                        ? 'number'
+                                        : 'text'
+                                }
+                                maxLength={
+                                    field['text-input-id'] === 'pregnancy-weeks'
+                                        ? 2
+                                        : null
+                                }
+                            />
+                        </List>,
+                    )
+                }
+
+                if (field.group != null) {
+                    const groupElements: ReactElement[] = []
+
+                    // add each group element to the internalElement array
+                    field.group.forEach((ele: any) => {
+                        if (ele.id != null && ele.name != null) {
```

Comment:
Do null values get added to the array if this line isn't there?

Reply:
With the current state of the JSON source file, no. In my mind it was a 'double check' to make sure that fields that don't exist for a particular entry are not accessed within the function. This would prevent runtime errors if invalid/incomplete data were added to the JSON in the future.

Reply:
Gotcha. From a dev standpoint I thought null values were coming out of that which confused me. I think a test can be written to check for null entries in the JSON as a possible alternative, but I'll leave it to your discretion to keep it this way or move it to a test method.

Reply:
"test to check for null entries in the JSON" <- Validating JSON data would be a good step. Currently we have no testing process for that + the pipeline of how devs / volunteers would edit the JSON + rebuild the website is unclear. Made a [ticket here for now](https://github.com/Haiti-Distributed-Healthcare-System/hdhs/issues/93), but like Zain, I think this can be resolved now @meghanmc